vininfo/details/toyota.py

```diff
@@ -0,0 +1,165 @@
+from ._base import VinDetails, Detail
+from ..dicts.bodies import *
+
+engine_prior_to_1996 = Detail(
+    ('vds', 0), {
+        '4': '7A-FE Lean Burn',
+        'A': '3MZ-FE',
+        'B': ['1NZ-FXE', '2AZ-FXE'],
+        'D': '2AZ-FE',
+        'E': '2AZ-FE',
+        'F': '1MZ-FE',
+        'G': '5S-FE',
+        'H': '1AZ-FE',
+        'K': '2GR-FE',
+        'L': '2RZ-FE',
+        'M': '3RZ-FE',
+        'N': '5VZ-FE',
+        'P': '3S-FE',
+        'R': '1ZZ-FE',
+        'S': '1BM',
+        'T': ['2UZ-FE', '1NZ-FE', '3S-GTE'],
+        'U': '1GR-FE',
+        'Y': '2ZZ-GE',
+    })
+
+engine_since_1996 = Detail(
+    ('vds', 1), {
+        'A': ['4A-FE', '7A-FE', '2AD-FTV (2005+)'],
+        'B': ['1HZ', '2AD-FHV (2005+)'],
+        'C': ['2C', '2CT', '2CT-E'],
+        'E': ['2JZ-GT', '2JZ-GTE', '2AZ-FE'],
+        'F': ['1MZ-FE', '2AR-FE'],
+        'H': ['1AZ-FE', '1NR-FE'],
+        'J': ['1FZ-FE', '1AZ-FSE'],
+        'N': ['5VZ-FE', '2ZR-FXE'],
+        'P': '2AZ-FSE',
+        'R': '1ZZ-FE',
+        'S': ['3S-FE', 'Electric (RAV4 EV only)'],
+        'T': '2UZ-FE',
+        'U': '2ZR-FE (Corolla Conquest 2010)',
+        'V': ['1NR-FE', '1VD-FTV'],
+        'W': ['2NZ-FE', '1CD-FTV'],
+        'X': '2TR-FE',
+        'Y': ['2ZZ-GE', '3UR-FE'],
+        'Z': ['3.5L 2GR-FKS V6 (278 hp)', '2JZ-FE', '1ZZ-FE', '3ZZ-FE'],
+    })
+
+
+class ToyotaDetails(VinDetails):
+    """Toyota VIN details extractor."""
+
+    # TODO: get year from VIN
+    # 1) Does the current design allow it, or should it evolve?
+    # 2) Add `most_likely_year` property to VIN?
+    year = 2010
```

Comment:
@idlesign Can you please help me with this?

Reply:
1) `VinDetail` has `_vin` property. So you can for example make model a @property which will return a proper Detail definition.

2) This needs to be thought over to design it properly. It needs rather to be a range (since-[optional till]). E.g. `VinDetails` can be supplied with `years` dynamic property. But optional till will increase the library maintenance cost. I'd rather not to mess with that.